src/snowflake/connector/pandas_tools.py

```diff
@@ -280,6 +294,27 @@ def write_pandas(
             stacklevel=2,
         )
 
+    # use_logical_type should be True when dataframe contains datetimes with timezone.
+    # https://github.com/snowflakedb/snowflake-connector-python/issues/1687
+    if not use_logical_type and any(
+        [pandas.api.types.is_datetime64tz_dtype(df[c]) for c in df.columns]
+    ):
+        warnings.warn(
+            "Dataframe contains a datetime with timezone column, but "
+            f"'use_logical_type = {use_logical_type}'. This can result in dateimes "
```

Comment:
There's a shortcut for this starting with Python 3.8 https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

Reply:
[fixed](https://github.com/snowflakedb/snowflake-connector-python/pull/1707/commits/fb84282b04e66f7decb7732b37e2001c2b0b3009)
also [synced with upstream](https://github.com/snowflakedb/snowflake-connector-python/pull/1707/commits/0798b1fdab15dd43930e717db47ca506cb7ffb6a)